client/src/partials/records/sales_records/sales_records.js

```diff
@@ -142,7 +142,12 @@ angular.module('bhima.controllers')
 
     function totalCost() {
       return $scope.model.sale.data.reduce(function (a, b) {
```

Comment:
var items = $scope.model.sale.data;
var total = items.filter(function (item) { return !item.creditId; })
.reduce(function (a, b) { return a + b.cost; }, 0);
return total;

This instruction is good, but not fast because the array will be covered twice, for the filter function and for the reduce function.